### PR TITLE
Docs: Fix broken links in Nodejs-api docs

### PR DESCRIPTION
**Pull request by jooohhn**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
- Changed links in `nodejs-api.md`. Should fix broken links in the [ESLint website docs](https://eslint.org/docs/developer-guide/nodejs-api)
- Related to website issue (https://github.com/eslint/website/issues/739)

### CODE REVIEW

File: docs/developer-guide/nodejs-api.md [hunk 1]
```diff
@@ -7,19 +7,19 @@ While ESLint is designed to be run on the command line, it's possible to use ESL
...
-    * [calculateConfigForFile()][eslint-calculateConfigForFile]
-    * [isPathIgnored()][eslint-isPathIgnored]
-    * [loadFormatter()][eslint-loadFormatter]
-    * [static version][eslint-version]
-    * [static outputFixes()][eslint-outputFixes]
-    * [static getErrorResults()][eslint-getErrorResults]
-    * [LintResult type](lintresult)
-    * [LintMessage type](lintmessage)
-    * [EditInfo type](editinfo)
-    * [Formatter type](formatter)
```
Comment (mysticatea) [inline docs/developer-guide/nodejs-api.md:22]: There are definitions of `[eslint-constructor]`-like links at the end of this file. Would you fix those instead? Those links are used in other places as well.

**Comment by mysticatea** [comment]
Hmm.

The page in GitHub ([/docs/developer-guide/nodejs-api.md](https://github.com/eslint/eslint/blob/7686d7feaccc7b8fee927eda6602d641d8de1e5c/docs/developer-guide/nodejs-api.md)) works fine. The root cause is, our website and GitHub markdown create different IDs for the markdown headers.

I wonder if we can fix our website's ID generator or not.

**Comment by jooohhn** [comment]
Updated table of contents to use new links at file end. It still has the problem you mentioned about Github docs and website docs having different header IDs. (can jump on website, but not on Github md)

**Comment by kaicataldo** [comment]
We use [AnchorJS](https://github.com/eslint/website/blob/master/src/js/main/index.js#L35) to create these links, if someone wants to see if we can get the behavior to match how GitHub generates these links.

**Comment by mdjermanovic** [comment]
We could override AnchorJS's [urlify](https://github.com/bryanbraun/anchorjs/blob/261c6d3e5013cc82a7682e427378f1dd89d8456d/anchor.js#L234), though I'm not sure if it's part of its public API.

Or, generate header ids upfront, which is [supported by AnchorJS](https://www.bryanbraun.com/anchorjs/#generating-navigations):

> You can also use AnchorJS alongside a static navigation with pre-defined IDs

Or, maybe use a different library that supports custom functions that create ids, instead of AnchorJS.

As for the function, there's [github-slugger](https://github.com/Flet/github-slugger). Though, I'm not sure if it matches what github is doing with that `◆` character we have on this page.

File: docs/developer-guide/nodejs-api.md [hunk 2]
```diff
@@ -8,18 +8,18 @@ While ESLint is designed to be run on the command line, it's possible to use ESL
...
+    * [static outputFixes()][eslint-outputfixes]
+    * [static getErrorResults()][eslint-geterrorresults]
+    * [LintResult type][lintresult]
+    * [LintMessage type][lintmessage]
+    * [EditInfo type][editinfo]
+    * [Formatter type][formatter]
```
Comment (mdjermanovic) [inline docs/developer-guide/nodejs-api.md:22]: :+1:

this was broken in markdown

**Comment by mdjermanovic** [comment]
It seems that the main problem should be fixed in the [eslint/website](https://github.com/eslint/website) repo, but I think we should merge this PR first, because it converts uppercase links to lowercase, which is consistent with other documents and most likely something we'll rely on. 

GitHub seems to have a script that handles any case, but we don't have that on the website. For example:

https://github.com/eslint/eslint/blob/master/docs/user-guide/configuring.md#Specifying-Processor (works)

https://eslint.org/docs/user-guide/configuring#Specifying-Processor (doesn't work)

Also, this PR fixes links that were broken for another reason (`LintResult type`, `LintMessage type`, `EditInfo type`, and `Formatter type`).

File: docs/developer-guide/nodejs-api.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ const { ESLint } = require("eslint");
 });
 ```
 
-### ◆ new ESLint(options)
+### new ESLint(options)
```
Comment (mdjermanovic) [inline docs/developer-guide/nodejs-api.md:98]: I'm not sure if we want to remove `◆` 

(though, removing these will probably lead to less possible problems later on the website)

Thoughts from the team?

File: docs/developer-guide/nodejs-api.md [hunk 4]
```diff
@@ -1382,17 +1382,17 @@ ruleTester.run("my-rule", myRule, {
...
-[formatter]: #-formatter-type
+[eslint-constructor]: #new-eslint-options
+[eslint-lintfiles]: #eslint-lintfiles-patterns
+[eslint-linttext]: #eslint-linttext-code-options
+[eslint-calculateconfigforfile]: #eslint-calculateconfigforfile-filepath
+[eslint-ispathignored]: #eslint-ispathignored-filepath
+[eslint-loadformatter]: #eslint-loadformatter-nameorpath
+[eslint-version]: #eslint-version
+[eslint-outputfixes]: #eslint-outputfixes-results
+[eslint-geterrorresults]: #eslint-geterrorresults-results
```
Comment (mdjermanovic) [inline docs/developer-guide/nodejs-api.md:1393]: These should be reverted to work well on GitHub (we should just convert uppercase to lowercase).

**Comment by mdjermanovic** [comment]
The broken links have been fixed, so closing.

**Pull request closed without merging**